sql/template_tf.go

```diff
@@ -0,0 +1,388 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+const tfTrainTemplateText = `
+import os
+# Disable Tensorflow INFO and WARNING logs
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+
+import sys, json
+import tensorflow as tf
+import functools
+try:
+    import sqlflow_models
+except:
+    pass
+
+from sqlflow_submitter.db import connect, db_generator
+
+# Disable Tensorflow INFO and WARNING
+import logging
+tf.get_logger().setLevel(logging.ERROR)
+
+BATCHSIZE = 1
+EPOCHS = None
+VERBOSE = 0
+
+train_args = dict()
+{{range $key, $value := .Attrs}}
+{{if eq $key "BATCHSIZE"}}
+BATCHSIZE = {{$value}}
+{{else if eq $key "EPOCHS"}}
+EPOCHS = {{$value}}
+{{else if eq $key "VERBOSE"}}
+VERBOSE = int({{$value}})
+{{else}}
+train_args["{{$key}}"] = {{$value}}
+{{end}}
+{{end}}
+
+driver="{{.Driver}}"
+{{if ne .Database ""}}
+database="{{.Database}}"
+{{else}}
+database=""
+{{end}}
+
+conn = connect(driver, database, user="{{.User}}", password="{{.Password}}", host="{{.Host}}", port={{.Port}})
+
+feature_columns = dict()
+{{ range $target, $colsCode := .FeatureColumnsCode }}
+feature_columns["{{$target}}"] = []
+{{ range $col := $colsCode }}
+feature_columns["{{$target}}"].append({{$col}})
+{{ end }}
+{{ end }}
+
+
+feature_column_names = [{{range .X}}
+"{{.FeatureName}}",
+{{end}}]
+
+
+classifier = {{.Estimator}}(
+    **feature_columns,
+    **train_args,
+    {{if .IsKerasModel}}
+)
+    {{else}}
+    model_dir = "{{.Save}}")
+    {{end}}
+
+{{/* Convert go side featureSpec to python dict for input_fn */}}
+feature_metas = dict()
+{{ range $value := .X }}
+feature_metas["{{$value.FeatureName}}"] = {
+    "feature_name": "{{$value.FeatureName}}",
+    "dtype": "{{$value.Dtype}}",
+    "delimiter": "{{$value.Delimiter}}",
+    "shape": {{$value.InputShape}},
+    "is_sparse": "{{$value.IsSparse}}" == "true"
+}
+{{end}}
+
+def get_dtype(type_str):
```

Comment:
shall we move `get_dtype`and `_parse_sparse_feature` to `sql/python`?

Reply:
`get_dtype` & `_parse_sparse_feature` are specified to `tensorflow`, I'm not sure if other templates would use this.